### PR TITLE
[New arch] Get folder content

### PR DESCRIPTION
**Pull request by abelgardep**
Implements #2956

### CODE REVIEW

**Review by theScrabi — APPROVED** [review]

File: owncloudApp/src/main/java/com/owncloud/android/dependecyinjection/UseCaseModule.kt [hunk 1]
```diff
@@ -27,8 +27,12 @@ import com.owncloud.android.domain.capabilities.usecases.GetCapabilitiesAsLiveDa
 import com.owncloud.android.domain.capabilities.usecases.GetStoredCapabilitiesUseCase
 import com.owncloud.android.domain.capabilities.usecases.RefreshCapabilitiesFromServerAsyncUseCase
 import com.owncloud.android.domain.files.usecases.CreateFolderAsyncUseCase
+<<<<<<< HEAD
```
Comment (theScrabi) [inline owncloudApp/src/main/java/com/owncloud/android/dependecyinjection/UseCaseModule.kt]: Make sure to remove these before a commit.
Please use 
`git diff HEAD` before committing.

Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/dependecyinjection/UseCaseModule.kt]: I know, rebase problems 😢  It was fixed in next commit

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    @Test(expected = Exception::class)
+    fun `get folder content - ko`() {
+        every { dao.getFolderContent(any()) } throws Exception()
+
+        localDataSource.getFolderContent(dummy.id!!)
+    }
+
+    @Test
+    fun `get folder images - ok`() {
```
Comment (theScrabi) [inline owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:126]: Why do you sometimes write `- ok` and sometimes `- ko`
Just a question of interest: what is the scheme for writing the description/names of the tests?

File: owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java [hunk 3]
```diff
@@ -226,13 +230,13 @@ private static String getDisplayNameFromContentResolver(Uri uri, Context context
         String mimeType = context.getContentResolver().getType(uri);
```
Comment (theScrabi) [inline owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java:230]: This method could be broken down into smaller ones.

File: owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java [hunk 4]
```diff
@@ -226,13 +230,13 @@ private static String getDisplayNameFromContentResolver(Uri uri, Context context
...
         if (mimeType != null) {
```
Comment (theScrabi) [inline owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java:231]: You could for example export this if clause.

**Review by theScrabi — CHANGES_REQUESTED** [review]
LFTM I like it. Yet there are smaller things. Maybe you can have a look at them. Sometimes I just have questions.

File: owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java (hunk 4 above)
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/utils/UriUtils.java:231]: Sure, but it is not the purpose of this PR, i just moved some hardcoded strings into constants. Probably we will convert to kotlin and refactor this file in upcoming PRs.

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt (hunk 2 above)
Comment (abelgardep) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:126]: I use `ok` as success and `ko` as exception or failure.

I started writing names as `getFolderContentSuccess` or `getFolderContentException` but it is not as simple when you have several cases. So i decided to switch to a more readable way:

what are we testing - expected behaviour - additional info

File: owncloudData/src/test/java/com/owncloud/android/data/file/repository/OCFileRepositoryTest.kt [hunk 5]
```diff
@@ -179,4 +179,28 @@ class OCFileRepositoryTest {
             localFileDataSource.getFileByRemotePath(OC_FILE.remotePath, OC_FILE.owner)
         }
     }
+
+    @Test
+    fun getFolderContentSuccess() {
+        every { localFileDataSource.getFolderContent(OC_FILE.parentId!!) } returns listOf(OC_FILE)
+
```
[the remark this answers is not in the document]
Comment (abelgardep) [reply owncloudData/src/test/java/com/owncloud/android/data/file/repository/OCFileRepositoryTest.kt:167]: I like to keep those statements separated, i think it is easy to read, don't you think? Something like this:

mock

function to test

result assertion

calls verification

**Review by theScrabi — APPROVED** [review]

**Pull request merged**